src/main/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookkeepingService.java

```diff
@@ -50,7 +49,7 @@ public HSMBookkeepingService(
                         bookkeepingConfig.getMaxAmountBlockHeaders(),
                         blockStore,
                         bookkeepingConfig.getDifficultyCap(),
-                        hsmVersion
+                        hsmBookkeepingClient.getVersion()
                 ),
                 nodeBlockProcessor,
                 bookkeepingConfig.getInformerInterval(),
```

Comment:
This is also strange, why pass the bookkeepingConfig object and also a value obtained from it?

Reply:
One Is Creating An Instance Of ConfirmedBlockHeadersProvider While The Other Is Used By The HSMBookkeepingService

Reply:
What I mean is that it passes `bookkeepingConfig.getInformerInterval()` and `bookkeepingConfig` to HSMBookkeepingService constructor. Shouldn't it get the advanceBlockchainTimeInterval from the bookkeepingConfig object? Or is there a reason to reason both values separately? Could the advanceBlockchainTimeInterval be provided form another source?

Reply:
HSMBookkeepingService Has 2 Constructors If You Take a Broader Look At The Class

Reply:
Created a New Task To Handle That.